Align with ESP8266 and fix signature deviation between Arduino.h and WMath.cpp

As per https://github.com/esp8266/Arduino/issues/8089, newer compilers flag the different types in Arduino.h and WMath.cpp for `makeWord` as an error. Proactively fix this for ESP32.
A patch is provided in PR #5230 

Edit: ESP32 master branch actually fails on this MCVE, too:
```
void setup()
{
    auto w = makeWord(1024);
    auto w2 = makeWord(192, 193);
}

void loop()
{
}
```

```
ld.exe: makeword.cpp.o:(.literal._Z5setupv+0x18): undefined reference to makeWord(unsigned short)

Error linking for board WEMOS D1 MINI ESP32
ld.exe: makeword.cpp.o: in function setup()
makeword.ino:3: undefined reference to makeWord(unsigned short)
Debug build failed for project 'makeword'
 
collect2.exe*: error: ld returned 1 exit status
```

--- a/cores/esp32/Arduino.h
+++ b/cores/esp32/Arduino.h
@@ -165,7 +165,7 @@ using std::min;
 using ::round;
 
 uint16_t makeWord(uint16_t w);
-uint16_t makeWord(byte h, byte l);
+uint16_t makeWord(uint8_t h, uint8_t l);
 
 #define word(...) makeWord(__VA_ARGS__)
 

--- a/cores/esp32/WMath.cpp
+++ b/cores/esp32/WMath.cpp
@@ -77,12 +77,12 @@ long map(long x, long in_min, long in_max, long out_min, long out_max) {
     return (delta * dividend + (divisor / 2)) / divisor + out_min;
 }
 
-unsigned int makeWord(unsigned int w)
+uint16_t makeWord(uint16_t w)
 {
     return w;
 }
 
-unsigned int makeWord(unsigned char h, unsigned char l)
+uint16_t makeWord(uint8_t h, uint8_t l)
 {
     return (h << 8) | l;
 }

